csv-validator-core/src/main/scala/uk/gov/nationalarchives/csv/validator/schema/Schema.scala

```diff
@@ -25,6 +25,8 @@ case class Quoted() extends GlobalDirective("quoted")
 
 case class TotalColumns(numberOfColumns: BigInt) extends GlobalDirective("totalColumns")
 
+case class IntegrityCheck(filepathColumn: String, includeFolder: Boolean) extends GlobalDirective("integrityCheck")
```

Comment:
@valydia I can't see your calling code so I don't yet really understand Your argument. Your metadata and closure files are different CSV files, and therefore you will have different CSV Schema files. 

I assume your code looks a bit like -

```scala
CSVValidator.create(options).validate(csvs, csv)
```

So you will be creating two validators anyway right? One for the metadata file and one for the closure file? If so, why do you also need the global directive? You could just pass the parameters for your integrity check in the options.